### ISSUE TITLE
Installation via pip hangs on OS X with Apple silicon

### ISSUE THREAD

**Issue by vsbuffalo** [MomentsLD/moments#165]
When using Python 3.12.2 on Apple silicon (M1 chip) I find that installation of moments hangs indefinitely. Here is the output:
```
❯ uv venv
Using Python 3.12.2 interpreter at: /Users/vsb/.pyenv/versions/3.12.2/bin/python3
Creating virtualenv at: .venv
Activate with: source .venv/bin/activate

~/src/moments-test
❯ source .venv/bin/activate

~/src/moments-test
.venv ❯ pip install mom

~/src/moments-test
.venv ❯  pip install moments@git+https://github.com/MomentsLD/moments.git
Collecting moments@ git+https://github.com/MomentsLD/moments.git
  Cloning https://github.com/MomentsLD/moments.git to /private/var/folders/4w/tx_sszv90dlbrx0thsygn_140000gn/T/pip-install-bdikswj2/moments_c789bf8805934ad39a2d5d480c35f2a6
  Running command git clone --filter=blob:none --quiet https://github.com/MomentsLD/moments.git /private/var/folders/4w/tx_sszv90dlbrx0thsygn_140000gn/T/pip-install-bdikswj2/moments_c789bf8805934ad39a2d5d480c35f2a6
  Resolved https://github.com/MomentsLD/moments.git to commit e379d9f0613a57b8d8a800321eb3a8b087f5f9b5
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.12.1 in /Users/vsb/.pyenv/versions/3.12.2/lib/python3.12/site-packages (from moments@ git+https://github.com/MomentsLD/moments.git) (1.26.4)
Collecting cython>=0.25 (from moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached Cython-3.0.10-py2.py3-none-any.whl.metadata (3.2 kB)
Collecting scipy>=1.3 (from moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached scipy-1.13.0-cp312-cp312-macosx_12_0_arm64.whl.metadata (60 kB)
Collecting mpmath>=1.0 (from moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting demes>=0.2 (from moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached demes-0.2.3-py3-none-any.whl.metadata (1.9 kB)
Collecting attrs>=20.3.0 (from demes>=0.2->moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached attrs-23.2.0-py3-none-any.whl.metadata (9.5 kB)
Collecting ruamel.yaml>=0.15.78 (from demes>=0.2->moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached ruamel.yaml-0.18.6-py3-none-any.whl.metadata (23 kB)
Collecting ruamel.yaml.clib>=0.2.7 (from ruamel.yaml>=0.15.78->demes>=0.2->moments@ git+https://github.com/MomentsLD/moments.git)
  Using cached ruamel.yaml.clib-0.2.8-cp312-cp312-macosx_13_0_arm64.whl.metadata (2.2 kB)
Using cached Cython-3.0.10-py2.py3-none-any.whl (1.2 MB)
Using cached demes-0.2.3-py3-none-any.whl (40 kB)
Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Using cached scipy-1.13.0-cp312-cp312-macosx_12_0_arm64.whl (30.4 MB)
Using cached attrs-23.2.0-py3-none-any.whl (60 kB)
Using cached ruamel.yaml-0.18.6-py3-none-any.whl (117 kB)
Using cached ruamel.yaml.clib-0.2.8-cp312-cp312-macosx_13_0_arm64.whl (135 kB)
Building wheels for collected packages: moments
  Building wheel for moments (pyproject.toml) ... -
```

Perhaps this is related to #129, but given that uses conda and I figured it's best to create a new issue. Note that this issue does *not* occur when I install moments on our Linux lab server.

**Comment by apragsdale** [comment]
Hi @vsbuffalo - thanks for the report. I haven't seen this before, but I also haven't used a Mac with those newer chips. I'll try to dig into this this week.

**Comment by vsbuffalo** [comment]
Hi All,
Here is an update — so I should have left it longer. It did complete eventually, taking about 12 minutes to compile. This is a bit odd since it's nearly instantaneous on Linux, but maybe it's using something pre-compiled in that case.

**Comment by apragsdale** [comment]
Thanks for checking that, Vince. I'm not sure why macos takes so much longer than linux, but my guess is it has to do with building they cython bits of the software. I would need to test them each independently.

Hopefully once #91 is finished soon, this won't be as much as a problem in the future.

**Comment by apragsdale** [comment]
I think we can close this (making note to look into the long installation times on macos). As of this evening, moments is installable via pip from wheels hosted on pypi: https://pypi.org/project/moments-popgen/

`pip install moments-popgen` will install.

Then you should be able to `import moments`.
